frontend/src/container/MetricsApplication/Tabs/Overview/ApDex/index.tsx

```diff
@@ -0,0 +1,50 @@
+import axios from 'axios';
+import Spinner from 'components/Spinner';
+import { Card, GraphContainer } from 'container/MetricsApplication/styles';
+import { useGetApDexSettings } from 'hooks/apDex/useGetApDexSettings';
+import { useNotifications } from 'hooks/useNotifications';
+import { memo } from 'react';
+import { useParams } from 'react-router-dom';
+
+import { IServiceName } from '../../types';
+import ApDexDataSwitcher from './ApDexDataSwitcher';
+import { ApDexApplicationProps } from './types';
+
+function ApDexApplication({
+	handleGraphClick,
+	onDragSelect,
+	topLevelOperationsRoute,
+	tagFilterItems,
+}: ApDexApplicationProps): JSX.Element {
+	const { servicename } = useParams<IServiceName>();
+	const { data, isLoading, error, isRefetching } = useGetApDexSettings(
+		servicename,
+	);
+	const { notifications } = useNotifications();
+
+	if (error && axios.isAxiosError(error)) {
+		notifications.error({
+			message: error.message,
+		});
+	}
+
+	return (
+		<Card>
+			<GraphContainer>
+				{isLoading || isRefetching ? (
```

Comment:
we can have condition to show the spinner above

Reply:
Okay